### ISSUE TITLE
use select for filter returning result but by default return all result

### ISSUE THREAD

**Issue by mortezakarimi** [ppetzold/nestjs-paginate#1025]
Current implementation needs to define select properties to use select in query params
I want to get all available properties but when I need to select some fields use `select` in query params and return selected properties
I have a big entity with many fields and relations and most of the time I need all results but in some cases like "select inputs" I just need the id and name, I cannot use the select param until I define explicitly select the property in the configuration

**Comment by Helveg** [comment]
Could you explain with code examples what the problem/behaviour is that you want?

**Comment by mortezakarimi** [comment]
```typescript
  static readonly PAGINATION_CONFIG: PaginateConfig<Project> = {
    sortableColumns: ['createdAt'],
    searchableColumns: ['name', 'serial'],
    nullSort: 'last',
    defaultSortBy: [['createdAt', 'DESC']],
    relations: [
      'a',
      'b',
      'a.a',
      'c',
      'c.a',
      'c.b',
      'd',
      'e',
      'f',
    ],
    filterableColumns: {},
  };

  async findAll(query: PaginateQuery) {
    return paginate(
      query,
      this.projectRepository,
      Service.PAGINATION_CONFIG,
    );
  }
```

currently, with the above config, we can get all information we need but in some cases we need to select only some properties. As you can see information we need is too much with relations so I can't specify `select` property in the pagination config

I want to use a query something like the following

```console
/api/projects?page=1&limit=20&select[]=id&select[]=name
```
to only select ID and name, but with the current implementation, we should specify selectable fields in the configuration to able to use select in query params

**Comment by Jarzebowsky** [comment]
This is something you can do with the usage of the `groups` on the `@Expose()` decorator.

**Comment by mortezakarimi** [comment]
> This is something you can do with the usage of the `groups` on the `@Expose()` decorator.

@Jarzebowsky Yes, you are right but there is a problem, I don't want to send another param to the backend from the frontend to specify this request use a specific group, also I don't want to write another API to use that group

When we already have the solution in the package why do we need to define it again using `@Expose` and `groups`

**Comment by Jarzebowsky** [comment]
You should define separate endpoints for returning just the data you want instead making this dynamic. The OpenAPI has it's schema which is a contract for your service requestors that you will return that kind of object with given data. If you are making this dynamic based on the request you are breaking this approach. What I would suggest to you is to define separate endpoint for the places you need different set of data OR you should return bigger number of data from your API anyway. To get just what you wanted you should move on with graphQL.
